File: src/main/java/ar/edu/unq/spring/persistence/PersonajeDAO.java
package ar.edu.unq.spring.persistence;

import ar.edu.unq.spring.modelo.Personaje;
import ar.edu.unq.spring.modelo.Personaje;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Collection;
import java.util.Optional;

public interface PersonajeDAO {
    Personaje save(Personaje personaje);
    Optional<Personaje> findById(Long id);
    Collection<Personaje> findAll();
    void deleteAll();
    void delete(Personaje personaje);
}

File: src/main/kotlin/ar/edu/unq/spring/persistence/PersonajeDAO.kt
package ar.edu.unq.spring.persistence

import ar.edu.unq.spring.modelo.Personaje
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.CrudRepository

interface PersonajeDAO : JpaRepository<Personaje, Long> {

    fun findByNombre(nombre:String): Personaje?
}
